parlai/core/search.py

```diff
@@ -0,0 +1,817 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
```

Comment:
This whole file looks like copypasta from another project (fairseq? hugging face?)

Is it really necessary? It's extremely large and seems to have a lot of redundancy with our other abstractions

Reply:
well, I've just taken this whole file from Parlai library because it contains LexicallyConstrainedBeamSearch. Basically we just need that class (and its parent Search class of course) but there are some more interesting Search sibling classes there like DiverseBeamSearch for example, which can be explored and implemented in Parlai later on as well. Also for the ease of updates. we can update the whole search.py file when required. But please let me know if you prefer to extract just the LexicallyConstrainedBeamSearch from it, we can do it as well.

Reply:
I definitely want to extract the very minimum. I have too much code to maintain as it is.

Reply:
I've trimmed search.py Please check now